src/languages/fsharp.js

```diff
@@ -154,15 +408,37 @@ export default function(hljs) {
         ]
       },
       {
-        scope: 'operator',
         // only non arithmetic operators that we can confidently match:
-        match: /\|{1,3}>|<\|{1,3}|->|<-|(?<!<)<<(?!<)|(?<!>)>>(?!>)|:>|:\?>?|\.\.|::|(?<!\|)\|(?!\|)/
+        scope: 'operator',
+        // split in two because we don't want the same relevance for all operators.
+        // warning: the order of the two modes is important, because |> collides with |
+        variants: [
+          {
+            // fsharp specific operators:
+            match: regex.either(
+              /\|{1,3}>/, // |> ||> |||>
+              /<\|{1,3}/, // <| <|| <|||
+              /(?<!<)<<(?!<)/, // << (but not < or <<<)
+              /(?<!>)>>(?!>)/, // >> (but not > or >>>)
+              /:>/,
+              /:\?>?/, // :?> :?
+            ),
+            relevance: 2
+          },
+          {
+            // operators common to many languages:
+            match: regex.either(
+              /->/,
+              /<-/,
+              /\.\./,
+              /::/,
+              /(?<!\|)\|(?!\|)/ // | (but not || or |||)
```

Comment:
We don't yet support negative lookbehind because not all browsers support it well yet.